File: today3/find-minimum-in-rotated-sorted-array.ts
/**
 * https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/
 */
export { findMin };

function findMin(nums: number[]): number {
  function bs(start: number, end: number): number {
    if (start >= end) return nums[start];

    const mid = Math.floor((start + end) / 2);
    if (nums[start] <= nums[mid]) {
      // Left portion of rotation
    } else {
      // Right portion of rotation
    }
  }
  return bs(0, nums.length - 1);
}

File: returned/python/today13/find_minimum_in_rotated_sorted_array.py
# https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/
from typing import List

class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        
        while l < r:
            m = l + (r - l) // 2
            if nums[m] > nums[r]:
                l = m + 1
            else:
                r = m
                
        return nums[l]
